### PR TITLE
kubelet: fix metric container_start_time_seconds's timestamp

### LINKED ISSUE #117880: kubelet resource metric container_start_time_seconds have timestamp equal to container start time
### What happened?

When scrape kubelet resource metrics provided by /metrics/resource, got response like this

```
curl -ik https://10.56.229.90:10250/metrics/resource --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt   --header "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
HTTP/2 200 
content-type: text/plain; version=0.0.4; charset=utf-8
date: Tue, 09 May 2023 08:46:01 GMT

# HELP container_cpu_usage_seconds_total [ALPHA] Cumulative cpu time consumed by the container in core-seconds
# TYPE container_cpu_usage_seconds_total counter
container_cpu_usage_seconds_total{container="calico-node",namespace="kube-system",pod="calico-node-9ftkg"} 13638.443124082 1683621961329
container_cpu_usage_seconds_total{container="controller",namespace="kube-system",pod="log-collector-t9htj"} 405.132439876 1683621961337
container_cpu_usage_seconds_total{container="kube-proxy",namespace="kube-system",pod="kube-proxy-nh5xv"} 2742.546427981 1683621961335
container_cpu_usage_seconds_total{container="nginx-proxy",namespace="kube-system",pod="nginx-proxy-10.56.229.90"} 150.481279171 1683621961338
container_cpu_usage_seconds_total{container="node-cache",namespace="kube-system",pod="nodelocaldns-mqm98"} 1087.07141331 1683621961332
container_cpu_usage_seconds_total{container="node-exporter",namespace="kube-system",pod="node-exporter-qpgk8"} 758.245076927 1683621961333
container_cpu_usage_seconds_total{container="otc-container",namespace="kube-system",pod="otel-collector-jwvkr"} 324.246997902 1683621961339
container_cpu_usage_seconds_total{container="speaker",namespace="kube-system",pod="lb-provisioner-speaker-5dpss"} 4506.641683615 1683621961331
# HELP container_memory_working_set_bytes [ALPHA] Current working set of the container in bytes
# TYPE container_memory_working_set_bytes gauge
container_memory_working_set_bytes{container="calico-node",namespace="kube-system",pod="calico-node-9ftkg"} 1.59567872e+08 1683621961329
container_memory_working_set_bytes{container="controller",namespace="kube-system",pod="log-collector-t9htj"} 3.2800768e+07 1683621961337
container_memory_working_set_bytes{container="kube-proxy",namespace="kube-system",pod="kube-proxy-nh5xv"} 3.1526912e+07 1683621961335
container_memory_working_set_bytes{container="nginx-proxy",namespace="kube-system",pod="nginx-proxy-10.56.229.90"} 1.7281024e+07 1683621961338
container_memory_working_set_bytes{container="node-cache",namespace="kube-system",pod="nodelocaldns-mqm98"} 2.8504064e+07 1683621961332
container_memory_working_set_bytes{container="node-exporter",namespace="kube-system",pod="node-exporter-qpgk8"} 2.4260608e+07 1683621961333
container_memory_working_set_bytes{container="otc-container",namespace="kube-system",pod="otel-collector-jwvkr"} 8.562688e+07 1683621961339
container_memory_working_set_bytes{container="speaker",namespace="kube-system",pod="lb-provisioner-speaker-5dpss"} 3.4607104e+07 1683621961331
# HELP container_start_time_seconds [ALPHA] Start time of the container since unix epoch in seconds
# TYPE container_start_time_seconds gauge
container_start_time_seconds{container="calico-node",namespace="kube-system",pod="calico-node-9ftkg"} 1.68327025431458e+09 1683270254314
container_start_time_seconds{container="controller",namespace="kube-system",pod="log-collector-t9htj"} 1.6832703827496119e+09 1683270382749
container_start_time_seconds{container="kube-proxy",namespace="kube-system",pod="kube-proxy-nh5xv"} 1.6832702508159482e+09 1683270250815
container_start_time_seconds{container="nginx-proxy",namespace="kube-system",pod="nginx-proxy-10.56.229.90"} 1.6832702502599943e+09 1683270250259
container_start_time_seconds{container="node-cache",namespace="kube-system",pod="nodelocaldns-mqm98"} 1.683270249690517e+09 1683270249690
container_start_time_seconds{container="node-exporter",namespace="kube-system",pod="node-exporter-qpgk8"} 1.6832702484719632e+09 1683270248471
container_start_time_seconds{container="otc-container",namespace="kube-system",pod="otel-collector-jwvkr"} 1.6836132802379303e+09 1683613280237
container_start_time_seconds{container="speaker",namespace="kube-system",pod="lb-provisioner-speaker-5dpss"} 1.6832702502273183e+09 1683270250227
# HELP node_cpu_usage_seconds_total [ALPHA] Cumulative cpu time consumed by the node in core-seconds
# TYPE node_cpu_usage_seconds_total counter
node_cpu_usage_seconds_total 86884.08430416 1683621957070
# HELP node_memory_working_set_bytes [ALPHA] Current working set of the node in bytes
# TYPE node_memory_working_set_bytes gauge
node_memory_working_set_bytes 4.05331968e+09 1683621957070
# HELP pod_cpu_usage_seconds_total [ALPHA] Cumulative cpu time consumed by the pod in core-seconds
# TYPE pod_cpu_usage_seconds_total counter
pod_cpu_usage_seconds_total{namespace="kube-system",pod="calico-node-9ftkg"} 0 1683621960955
pod_cpu_usage_seconds_total{namespace="kube-system",pod="kube-proxy-nh5xv"} 0 1683621944751
pod_cpu_usage_seconds_total{namespace="kube-system",pod="lb-provisioner-speaker-5dpss"} 0 1683621947023
pod_cpu_usage_seconds_total{namespace="kube-system",pod="log-collector-t9htj"} 0 1683621958262
pod_cpu_usage_seconds_total{namespace="kube-system",pod="nginx-proxy-10.56.229.90"} 0 1683621958602
pod_cpu_usage_seconds_total{namespace="kube-system",pod="node-exporter-qpgk8"} 0 1683621946994
pod_cpu_usage_seconds_total{namespace="kube-system",pod="nodelocaldns-mqm98"} 0 1683621956580
pod_cpu_usage_seconds_total{namespace="kube-system",pod="otel-collector-jwvkr"} 0 1683621961292
# HELP pod_memory_working_set_bytes [ALPHA] Current working set of the pod in bytes
# TYPE pod_memory_working_set_bytes gauge
pod_memory_working_set_bytes{namespace="kube-system",pod="calico-node-9ftkg"} 0 1683621960955
pod_memory_working_set_bytes{namespace="kube-system",pod="kube-proxy-nh5xv"} 0 1683621944751
pod_memory_working_set_bytes{namespace="kube-system",pod="lb-provisioner-speaker-5dpss"} 0 1683621947023
pod_memory_working_set_bytes{namespace="kube-system",pod="log-collector-t9htj"} 0 1683621958262
pod_memory_working_set_bytes{namespace="kube-system",pod="nginx-proxy-10.56.229.90"} 0 1683621958602
pod_memory_working_set_bytes{namespace="kube-system",pod="node-exporter-qpgk8"} 0 1683621946994
pod_memory_working_set_bytes{namespace="kube-system",pod="nodelocaldns-mqm98"} 0 1683621956580
pod_memory_working_set_bytes{namespace="kube-system",pod="otel-collector-jwvkr"} 0 1683621961292
# HELP scrape_error [ALPHA] 1 if there was an error while getting container metrics, 0 otherwise
# TYPE scrape_error gauge
scrape_error 0
```

### What did you expect to happen?

The metric container_start_time_seconds's timestamp should not equals to container start time

### How can we reproduce it (as minimally and precisely as possible)?

As above

### Anything else we need to know?

_No response_

### Kubernetes version

v1.24.7

### Cloud provider

Volc Engine

### OS version

_No response_

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by philchia**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
fix #117880 

#### Which issue(s) this PR fixes:
Fixes #117880

### CODE REVIEW

File: pkg/kubelet/metrics/collectors/resource_metrics.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"k8s.io/component-base/metrics"
 	"k8s.io/klog/v2"
 	summary "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+
```
Comment (bart0sh) [inline pkg/kubelet/metrics/collectors/resource_metrics.go:26]: Can you remove this change? I don't think it's related to the PR.

Comment (philchia) [reply pkg/kubelet/metrics/collectors/resource_metrics.go]: done

**Comment by philchia** [comment]
@bart0sh PTAL

**Review by logicalhan — APPROVED** [review]
/triage accepted
/cc @dgrisonnet 
/lgtm
/approve
/hold

**Comment by dgrisonnet** [comment]
Looks like `pull-kubernetes-node-crio-e2e` is indeed failing.

IMO we should try to fix that test to not require adding a collection timestamp to `container_start_time_seconds` since it doesn't really have one.

**Comment by saschagrunert** [comment]
Ah, yeah. @dgrisonnet are you still working on this?

**Comment by dgrisonnet** [comment]
I am not

**Review by dashpole — APPROVED** [review]
/lgtm
Thanks for catching this!

**Comment by dashpole** [comment]
> This seems like a revert of https://github.com/kubernetes/kubernetes/pull/103429 which will reintroduce the test failures from https://github.com/kubernetes/kubernetes/issues/103428.

Ah, that makes sense.  Looking

**Comment by dashpole** [comment]
Can we change the test here: https://github.com/kubernetes/kubernetes/blob/1a1a3d0a87530aea0294ce7ce91a79e7378b95a7/test/e2e_node/resource_metrics_test.go#L165

To treat a zero/non-existent timestamp as time.Now()?

**Comment by saschagrunert** [comment]
Attemtping a carry-over in https://github.com/kubernetes/kubernetes/pull/120518

**Pull request closed without merging**